### PR TITLE
Add scaling operator

### LINKED ISSUE #3766: Scaling Operator

### PR DESCRIPTION
**Pull request by kuke**
Resolve #3766

### CODE REVIEW

File: paddle/operators/scaling_op.cc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    AddInput("weight", "The weight vector of scaling operator.");
+    AddOutput("Out", "The output tensor of scaling operator.");
+    AddComment(R"DOC(Scaling operator
+
+The equation is: Out.row[i] = weight[i] * X.row[i]
+)DOC");
+  }
+};
+
+class ScalingGradOp : public framework::OperatorWithKernel {
```
Comment (reyoung) [inline paddle/operators/scaling_op.cc]: The gradient operator could be composed by two forward operator. See `minus_op`.

File: python/paddle/v2/framework/tests/test_scaling_op.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            'X': np.random.random((32, 64)).astype("float32"),
+            'weight': np.random.random(32).astype("float32")
+        }
+        self.check_grad(
+            op, inputs, set(['X', "weight"]), "Out", max_relative_error=0.5)
+
+
+if __name__ == '__main__':
+    unittest.main()
+if __name__ == '__main__':
```
Comment (reyoung) [inline python/paddle/v2/framework/tests/test_scaling_op.py]: Maybe __main__ should not be defined twice.

**Comment by qingqing01** [comment]
> Maybe it should be named colwise_mul?

@reyoung Discussed with @kuke and @gongweibao , we'd like to merge this op with elementwise mul op.

**Comment by kuke** [comment]
@reyoung @qingqing01 thanks for your comments. I will work with @gongweibao to unify these mul ops.

**Comment by kuke** [comment]
Closed this PR for #3787 has a more complete implementation.

**Pull request closed without merging**